File: TaskManager/TaskManager/IssueRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TaskManager
{
    public class IssueRepository
    {
        private Dictionary<int, Issue> issues = new Dictionary<int, Issue>();

        public void AddIssue(Issue issue)
        {
            issues[issue.Id] = issue;
        }

        public Issue GetIssue(int id)
        {

            if(issues.ContainsKey(id))
            {
                return issues[id];
            }

            else
            {
                return null;
            }
        }   

        public void UpdateIssue(Issue issue)
        {
            if (issues.ContainsKey(issue.Id))
            {
                issues[issue.Id] = issue;
            }
        }

        public void DeleteIssue(int id)
        {
            issues.Remove(id);
        }

        public List<Issue> GetAllIssues()
        {
            return new List<Issue>(issues.Values);
        }
    }
}

File: TaskManager/TaskManagerSolution/src/IssueRepository.java
import java.util.HashMap;
import java.util.Map;

public class IssueRepository {
    private final Map<String,Issue> issues = new HashMap<>();

    public void addIssue(Issue issue) {
        issues.put(issue.getId(), issue);
    }

    public Issue getIssue(String id) {
        return issues.get(id);
    }

    public void removeIssue(String id) {
        issues.remove(id);
    }

    public void updateIssue(Issue issue) {
        if (issues.containsKey(issue.getId())) {
            issues.put(issue.getId(), issue);
        }
    }
}
